src/screens/CreateContactScreen.tsx

```diff
@@ -36,6 +37,15 @@ export class CreateContactScreen extends React.PureComponent<Props, State> {
     error: '',
   };
 
+  static getDerivedStateFromProps(props: Props, state: State) {
```

Comment:
I think it would be cleaner and easier to read only address from route props so we will have one source of truth. So instead of setting  local state we can use navigation.setParams at line 73

Reply:
I don't get it.

Reply:
I would:
1. Remove address from state
2. Read only address from props.route.params.address
3. Set address by the usage of function https://reactnavigation.org/docs/navigation-prop/#setparams

Reply:
What if the user wants to provide the address by himself. Beside that address from param (if provided) would not be seen in input field.

Reply:
user can still provide an address by himself by setting params like this
 
setAddress = (address: string) => {
const { navigation } = this.props
navigation.setParams({ address });
}

Reply:
But in normal flow, you get into this screen without any address in param so setting up not existing param will cause a crash. I don't think that this is either a cleaner or easier way,

Reply:
you have something like initialParams where you can set address to ''

Reply:
This is not even consistent with the rest of the app. Does your proposition give us any performance / technical pros or is it just opinion? Everything can be done with plenty of ways

Reply:
Agreed but generally react team doesn't encourage to use derived state (https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html) if absolutely necessary. Consider this:
1. initialParams is like setting state in constructor
2. setParams is like using setState
3. reading from this.props.route.params is like reading from this.state
This all is intended to use in this kind of situation